File: app/src/main/java/com/example/cb/test/mvp/MvpActivity.java
package com.example.cb.test.mvp;


import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

import cb.xlibrary.utils.XLogUtils;
import com.example.cb.test.R;
import com.example.cb.test.base.BaseMvpActivity;
import com.example.cb.test.mvp.presenter.NewListPresenter;
import com.example.cb.test.mvp.view.INewListView;
import com.example.cb.test.rx.http.NewsResp;

public class MvpActivity extends BaseMvpActivity<NewListPresenter> implements INewListView {

    private FragmentManager fragmentManager;
    private MvpFragment mvpFragment;

    @Override
    protected NewListPresenter createPresenter() {
        return new NewListPresenter(this);
    }

    @Override
    protected int getContentViewId() {
        return R.layout.activity_mvp;
    }

    @Override
    public void initView() {
        if (fragmentManager == null) {
            fragmentManager = getSupportFragmentManager();
        }

        mvpFragment = new MvpFragment();

        FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.add(R.id.content, mvpFragment, "mvpFragment");
        transaction.commit();

    }

    @Override
    public void initData() {
        // mPresenter.fetchData();
    }

    @Override
    public void initListener() {


    }

    @Override
    public void onRequestSuccess(NewsResp data) {
        XLogUtils.d(data.getResult().getData().get(0).getAuthor_name());
    }

    @Override
    public void onError(String error) {
        XLogUtils.e(error);
    }


}

File: app/src/main/java/com/example/cb/test/mvp/MvpActivity.kt
package com.example.cb.test.mvp

import androidx.fragment.app.FragmentManager
import com.example.cb.test.R
import com.example.cb.test.base.BaseMvpActivity
import com.example.cb.test.mvp.presenter.NewListPresenter
import com.example.cb.test.mvp.view.INewListView
import com.example.cb.test.rx.http.NewsResp

/**
 * 练习MVP
 */
class MvpActivity : BaseMvpActivity<NewListPresenter>(), INewListView {


    private var fragmentManager: FragmentManager? = null
    private var mvpFragment: MvpFragment? = null


    override fun createPresenter(): NewListPresenter {
        return NewListPresenter(this)
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_mvp
    }

    override fun initUI() {
        if (fragmentManager == null) {
            fragmentManager = supportFragmentManager
        }
        //创建mvpFragment对象
        mvpFragment = MvpFragment()

        val transaction = fragmentManager!!.beginTransaction()
        transaction.add(R.id.content, mvpFragment!!, "MvpFragment")
        transaction.commit()

        mPresenter!!.fetchData()
    }

    override fun initEvent() {
    }

    override fun onRequestSuccess(data: NewsResp) {
        toast(data.result.data[0].author_name)
    }

    override fun onError(error: String) {
        toast(error)
    }
}